zulipterminal/ui_tools/boxes.py

```diff
@@ -1112,22 +1112,34 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
         if not reactions:
             return ""
         try:
-            reaction_stats = defaultdict(set)
+            my_user_id = self.model.user_id
+            reaction_stats = defaultdict(list)
```

Comment:
Let's maybe do the `set` to `list` change in a refactor commit before everything else, so that the diffs look cleaner?

Reply:
This has turned into a very small commit, but we can always squash it later, and generally small is good for reviews too. However, let me address the ordering :)

I know Rohitth said "before everything else", but note that the original 'story' of commits was to adjust the spacing and then add the user names. Putting this first feels strange to a reviewer as this seems unrelated to the next commit (spacing), and maybe part of a new 'feature' change, ie. adjusting the ordering?

If there are purely refactoring commits these generally come before the related feature. This is currently a pure refactor since it doesn't change the behavior, but it seems unrelated to the following commit. This is easy to reorder in this case since the changes don't intersect or depend on each other (you can move the commits around using an interactive rebase).

The ordering change is arguably a new (visible) feature, so you might consider this part of that block of commits.

Other than making it easier to review, this also helps with merging earlier commits (or groups of commits) from a PR first, if PRs become unexpectedly large, or if earlier commits are mergeable and later ones need further work.